doc/deps.md

```diff
@@ -180,10 +180,10 @@ avoid using them if it's not necessary.
 
 ```bash
 # Download and uncompress Thrift:
-wget "http://www.apache.org/dyn/mirrors/mirrors.cgi?action=download&filename=thrift/0.13.0/thrift-0.13.0.tar.gz" \
-  -O thrift-0.13.0.tar.gz
-tar -xvf ./thrift-0.13.0.tar.gz
-cd thrift-0.13.0
+wget "http://www.apache.org/dyn/mirrors/mirrors.cgi?action=download&filename=thrift/0.16.0/thrift-0.16.0.tar.gz" \
+  -O thrift-0.16.0.tar.gz
+tar -xvf ./thrift-0.16.0.tar.gz
+cd thrift-0.16.0
```

Comment:
The documentation needs further update, because now for Ubuntu 22.04, Thrift can be installed from the official package repositories, while for Ubuntu 20.04, manual compilation is required.

Reply:
Sorry I omitted these in the first review round.

Reply:
I did a somewhat larger refactoring here, that will even help with #631 .

I have reorganized the script folder structure in a way that now it supports compilation with catching abilities. Furthermore, I updated the CI accordingly. The documentation is now also changed.

Reply:
Nice caching solution. I have added some remarks, most of them are minor.